src/main/java/codeu/controller/ServerStartupListener.java

```diff
@@ -31,6 +34,9 @@ public void contextInitialized(ServletContextEvent sce) {
       List<Message> messages = PersistentStorageAgent.getInstance().loadMessages();
       MessageStore.getInstance().setMessages(messages);
 
+      HashMap<String, Hashtag> hashtags = PersistentStorageAgent.getInstance().loadHashtags();
```

Comment:
In the same way that you use `List`, which is an interface, instead of `ArrayList`, you should use `Map` instead of `HashMap`.  So you can basically find/replace `HashMap` with `Map` everywhere, with the exception of places where you say `new HashMap` (since `HashMap` is the implementation of the `Map` interface).

Reply:
Interesting! Fixed